### DISCUSSION TITLE
ContractFunctionExecutionError: An unknown error occurred while executing the contract function

### DISCUSSION THREAD

**Discussion by wmahmood1984** [WalletConnect/walletconnect-monorepo#2397] [Developer Support] [answered]
Hi, all my read and write functions are working perfectly in development environment. however, none of them works in production environment. Below is the error snippet. Please help me 

ContractFunctionExecutionError: An unknown error occurred while executing the contract function "totalAmountStaked".

Raw Call Arguments:
  to:    0xF55D41F33737bb00757A64deEB924e921B1Cf9f5
  data:  0xfe961f61
 
Contract Call:
  address:   0xF55D41F33737bb00757A64deEB924e921B1Cf9f5
  function:  totalAmountStaked()

Docs: https://viem.sh/docs/contract/readContract.html
Details: Cannot convert a BigInt value to a number
Version: viem@0.3.24
    at pe (http://localhost:3000/static/js/main.54649b31.js:2:3366915)
    at http://localhost:3000/static/js/main.54649b31.js:2:3376120
    at d (http://localhost:3000/static/js/main.54649b31.js:2:3189320)
    at Generator.<anonymous> (http://localhost:3000/static/js/main.54649b31.js:2:3190663)
    at Generator.throw (http://localhost:3000/static/js/main.54649b31.js:2:3189683)
    at r (http://localhost:3000/static/js/main.54649b31.js:2:3183000)
    at u (http://localhost:3000/static/js/main.54649b31.js:2:3183239)

**Comment by xzilja** [comment, marked as answer]
Looks like viem related issue, so best ask there

**Reply by wmahmood1984** [reply]
I think viem is just to convert the data into different types at front end. I am not getting data at the fetch level.

**Reply by finessevanes** [reply]
Could you please provide your implementation code?

**Reply by wmahmood1984** [reply]
hi, sorry just saw your message
how you want the implementation code? should i send you the file?
